### DISCUSSION TITLE
Arrange for the interpreter to ignore the #! in the first line

### DISCUSSION THREAD

**Discussion by pentalive** [robhagemans/pcbasic#216] [Ideas]
If somehow the interpreter would ignore #! `which pcbasic` as the first line - PC basic programs would become click launchable.  On should then be able to just find MYPROG.BAS in the file manager of one's choice and double click it - pcbasic launches with the MYPROG.BAS as a command line argument, and the basic program there in runs. If the basic program ends with a SYSTEM statement, then the app becomes just like any other app on that Linux box. 

When saving a program it should add that #! above.

**Comment by robhagemans** [comment]
Hi, thanks for the suggestion - I don't think I'll implement this though, for a few reasons
- it breaks compatibility with GW-BASIC - any program not starting with a line number should fail with a `Direct statement in file`
- the shebang line is for shell-scripts, and the GW-BASIC language is really quite unsuitable for shell scripting
- it's a bit of a niche, OS-specific feature, which means it's a pain for documentation, support and maintenance

**Comment by pentalive** [comment]
Hey Rob!

Thanks for your rely.  I understand totally - It's great to have GWBASIC again on my Linux Box.


-------------------------------------------------------------------------------------------------
Ronald Hudson.

Sent from my Linux Laptop

________________________________
